src/main/java/Robots/samples/ObstacleAvoidRobot.java

```diff
@@ -54,6 +55,33 @@ public void loop() throws Exception {
             } else {
                 motion.move(defaultMoveSpeed, defaultMoveSpeed, 1000);
             }
+        }*/
+
+        if(state == robotState.RUN){
+            double dist = distSensor.getDistance();
+            System.out.println("dist  3 " +distSensor.getDistance());
+
+
+            if(dist<=10){
+                if(dist<=1){
+                    motion.rotateDegree((int) (defaultMoveSpeed * 0.75), 91);
+                    if(distSensor.getDistance()<=1){
+                        motion.rotateDegree((int) (defaultMoveSpeed * 0.75), -91);
+                        motion.rotateDegree((int) (defaultMoveSpeed * 0.75), -91);
+                        if(distSensor.getDistance()<=1){
+                            motion.rotateDegree((int) (defaultMoveSpeed * 0.75), -91);
+                        }
+                    }
+                }else{
+                    motion.move(defaultMoveSpeed, defaultMoveSpeed, 100);
```

Comment:
Since the cell distances are a fixed value, can you try `motion.moveDistance()` function for it?